### PR TITLE
[CPU][ARM] Enable SLT Interpolate tests on ARM

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
 - Interpolate SLT test file has been splitted into test class and test instances.
 - As were agreed with @EgorDuplensky there are dedicated test instances for x64 and arm since ACL supports very limited scope of Interpolate configuration.
 - While enabling tests on ARM it was found that ACL NEScale does not support scales mode with linear/linear_onnx and half_pixel/asymmetric, so additional limitation has been added.
 - To add the limitation mentioned above `shapeCalcMode` was moved from `Interpolate` node into `InterpolateAttrs` structure.
 - Enabled `fp16` in arm test instances
 - Enabled `NHWC` in arm test instances

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 1]
```diff
@@ -1564,6 +1564,7 @@ size_t InterpolateKey::hash() const {
 
     size_t seed = 0;
 
+    seed = hash_combine(seed, nodeAttrs.shapeCalcMode);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1567]: Why you added it into hash computation? I believe JIT kernel doesn't depend on it

Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: reverted this line

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Comparison operator should be supported as well

image

Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: addressed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**